### PR TITLE
Task #1607: Apply New Styles to Package Documentation Module Navigation

### LINKED ISSUE #1607: Apply New Styles to Package Documentation Module Navigation
This is an Outreachy Medium difficulty issue. You must have completed a simple issue (marked with label "outreachy") first, to be eligible for being assigned to this one.

Please only request to be assigned this issue if you're applying to the Dark Mode Outreachy project.

# Task

Currently, the module navigation on the package documentation area (e.g. at https://ocaml.org/p/irmin/latest/doc/index.html) looks like this:
Screenshot 2023-10-10 at 17-45-14 irmin 3 8 0 (latest) · OCaml Package

New styles:
Screenshot from 2023-10-10 17-44-17

[Here's a Figma Document with the styles](https://www.figma.com/file/J0w4b9exaHhVpvabKoGqTn/Design-Github-Issues?type=design&node-id=27%3A136&mode=design&t=KUzqK4w2r2nJuXGF-1).

How to solve this issue:
1. Inspect the styles from the Figma document to adjust the styles of the different existing tags in `doc.css`.
2. Make changes in `navmap.eml` to use the open/close icons and implement the rest of the Figma design.

### PR DESCRIPTION
**Pull request by henilGondalia**
Resolves #1607

This PR addresses issue #1607, focusing on applying new style to Package Documentation Module Navigation of the OCaml.org website. [Here's a Figma reference design](https://www.figma.com/file/J0w4b9exaHhVpvabKoGqTn/Design-Github-Issues?type=design&node-id=27%3A136&mode=design&t=KUzqK4w2r2nJuXGF-1). 

## Screenshots of New Style

Doc Ocaml 1

Doc Ocaml 3

Please review the changes and let me know if any further adjustments are needed. Thank you!

### CODE REVIEW

**Comment by sabine** [comment]
Hey that looks really beautiful!

Here's some points to watch out for:

1. scroll bar overlapping +/- symbols
Screenshot from 2023-10-16 10-25-22

2. The three parameters here are at the same level in the hierarchy, but their left margin is different (see https://ocaml.org/p/irmin/3.8.0/doc/Irmin/Storage/Append_only/argument-2-K/index.html):
Screenshot from 2023-10-16 10-27-49

3. Click area for +/- symbols is very small (I recommend adding horizontal padding):
Screenshot from 2023-10-16 10-30-07

**Comment by henilGondalia** [comment]
Thank you, @sabine for appreciation. I'll work on implementing the suggested changes and fixing bugs.

**Comment by henilGondalia** [comment]
Hello @sabine as you suggested I have added some horizonal space for +/- and also to give some space from the scrollbar. Also fixed space issue in same level in the hierarchy.
Please review these updates and let me know your feedback.

**Comment by sabine** [comment]
I'm impressed, this looks really great to me @henilGondalia! Thanks for the contribution!

I'm putting this on staging.ocaml.org for others to test as well.

File: asset/css/doc.css [hunk 1]
```diff
@@ -34,7 +34,7 @@ div.odoc .spec a.anchor {
   position: absolute;
   left: 0;
   top:0.3em;
-  opacity: 0;
+  opacity: 1;
```
Comment (sabine) [inline asset/css/doc.css]: this change seems unrelated to the package documentation navigation

File: asset/css/doc.css [hunk 2]
```diff
@@ -47,19 +47,24 @@ div.odoc .spec a.anchor::after {
...
+/* div.odoc *:hover > a.anchor {
   opacity: 1;
 }
 
 div.odoc *:hover > a.anchor:hover {
   color: rgb(75 85 99);
+} */
+
+div.odoc *>a.anchor {
+  color: rgb(75 85 99);
```
Comment (sabine) [inline asset/css/doc.css]: seems unrelated to the package navigation documentation navigation

File: asset/css/doc.css [hunk 3]
```diff
@@ -201,9 +206,10 @@ div.odoc .comment-delim {
   cursor: pointer;
 }
 
-span.icon-expand:not(.open) > .navmap-tag , span.no-expand > .navmap-tag {
-  background:white;
-}
+/* span.icon-expand:not(.open)>.navmap-tag,
+span.no-expand>.navmap-tag {
+  background: white;
+} */
```
Comment (sabine) [inline asset/css/doc.css]: remove, instead of commenting

File: asset/css/doc.css [hunk 4]
```diff
@@ -273,30 +285,43 @@ span.icon-expand:not(.open) > .navmap-tag , span.no-expand > .navmap-tag {
   border-color: rgb(32, 68, 165);
 }
 
-span.no-expand > .navmap-tag {
-  margin-left: 12px;
+span.icon-expand>.navmap-tag,
+span.no-expand>.navmap-tag {
```
Comment (sabine) [inline asset/css/doc.css]:
```suggestion
span.icon-expand > .navmap-tag,
span.no-expand > .navmap-tag {
```

for better readability

**Review by sabine — COMMENTED** [review]
I noticed one last very little thing:

Screenshot from 2023-10-17 13-25-35

At desktop screen size, the little indicator line left of "lib" is cut off a bit and thus looks thinner than the others.

File: asset/css/doc.css (hunk 2 above)
Comment (henilGondalia) [reply asset/css/doc.css]: Initially, I thought it was related to the hover CSS of navigation line, so I made changes to it. However, upon checking again, I realized that it wasn't that css, as you rightly pointed out. It is a mistake which I will revert with lib indicator css issue.

**Comment by henilGondalia** [comment]
Hello @sabine,
Is there any other task that I can take on ?

**Comment by sabine** [comment]
@henilGondalia there's more applicants than issues still. I have ideas for two more medium issues I can set up tomorrow, but should probably assign folks that didn't have a chance to try a medium issue yet.

File: asset/css/doc.css [hunk 5]
```diff
@@ -47,7 +47,7 @@ div.odoc .spec a.anchor::after {
   padding: 0.6em;
 }
 
-div.odoc *:hover > a.anchor {
+ div.odoc *:hover > a.anchor {
```
Comment (sabine) [inline asset/css/doc.css]:
```suggestion
div.odoc *:hover > a.anchor {
```

**Review by sabine — APPROVED** [review]
Thanks for the contribution @henilGondalia! Excellent work!

**Comment by tmattio** [comment]
I absolutely love the new package nav UI! Thank you for your work @henilGondalia!

**Comment by henilGondalia** [comment]
> @henilGondalia there's more applicants than issues still. I have ideas for two more medium issues I can set up tomorrow, but should probably assign folks that didn't have a chance to try a medium issue yet.

Sure, if there's no one else available, I'd be happy to take on the task.

> Thanks for the contribution @henilGondalia! Excellent work!

Thank you @sabine for appreciation. It was a pleasure to contribute! 😊

**Comment by henilGondalia** [comment]
> I absolutely love the new package nav UI! Thank you for your work @henilGondalia!

Thank you @tmattio. I'm glad that you love it.
